src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs

```diff
@@ -187,81 +176,111 @@ await using (var ms = new MemoryStream())
+                        }
+                        catch (Exception e)
+                        {
+                            await DebugInfoPackageUtils.WriteExceptionAsZipEntryAsync(e, archive, $"Node - [{ServerStore.NodeTag}]");
+                        }
                     }
 
                     ms.Position = 0;
-                    await ms.CopyToAsync(ResponseBodyStream(), token.Token);
+                    await ms.CopyToAsync(ResponseBodyStream());
                 }
 
                 return null;
             }, operationId, token: token);
         }
 
-        private static async Task WriteLogFile(ZipArchive archive, CancellationToken token = default)
+        private static async Task WriteLogFile(ZipArchive archive, CancellationToken token)
         {
-            token.ThrowIfCancellationRequested();
-
-            var prefix = $"{_serverWidePrefix}/{DateTime.UtcNow:yyyy-MM-dd H:mm:ss}.txt";
-
-            try
+            var prefix = $"{_serverWidePrefix}/{DateTime.UtcNow:yyyy-MM-dd H:mm:ss}.log";
+            var entry = archive.CreateEntry(prefix, CompressionLevel.Optimal);
+            entry.ExternalAttributes = (int)(FilePermissions.S_IRUSR | FilePermissions.S_IWUSR) << 16;
+            await using (var entryStream = entry.Open())
+            await using (var sw = new StreamWriter(entryStream))
             {
-                var entry = archive.CreateEntry(prefix, CompressionLevel.Optimal);
-                entry.ExternalAttributes = ((int)(FilePermissions.S_IRUSR | FilePermissions.S_IWUSR)) << 16;
-
-                await using (var entryStream = entry.Open())
+                try
                 {
-                    LoggingSource.Instance.AttachPipeSink(entryStream);
+                    token.ThrowIfCancellationRequested();
 
+                    LoggingSource.Instance.AttachPipeSink(entryStream);
+                    
                     await Task.Delay(15000, token);
+                }
+                catch (Exception e)
+                {
+                    await sw.WriteAsync($"{DateTime.UtcNow.Add(new TimeSpan(LoggingSource.LocalToUtcOffsetInTicks)):yyyy-MM-ddTHH:mm:ss.fffffffZ}, {e.Message}");
+                }
+                finally
+                {
                     LoggingSource.Instance.DetachPipeSink();
-
-                    await entryStream.FlushAsync(token);
+                    await entryStream.FlushAsync();
                 }
             }
-            catch (OperationCanceledException)
+        }
+
+        private async Task<Stream> GetDebugInfoFromNodeAsync(JsonOperationContext context, RequestExecutor requestExecutor, long operationId, OperationCancelToken token, int? timeoutInSec)
+        {
+            var rawStreamCommand = new GetRawStreamResultCommand($"/admin/debug/info-package?operationId={operationId}&timeoutInSec={timeoutInSec}");
+            
+            var requestExecutionTask = requestExecutor.ExecuteAsync(rawStreamCommand, context);
```

Comment:
why not to pass a CT here?

Reply:
and run `KillOperation` if it will throw

Reply:
We want to finish the operation gracefully.
If you pass the `cts` here it will throw and we won't be able to get any result.

Reply:
Okay, now I understand... 
I think `timeoutInSec + 2 minutes` would be much simpler here.

Reply:
Also, would be great to show some progress (single node / cluster-wide), not sure if this should be part of this issue though.